plugins/auth-backend/src/lib/flow/authFlowHelpers.ts

```diff
@@ -25,16 +25,43 @@ export const postMessageResponse = (
 ) => {
   const jsonData = JSON.stringify(response);
   const base64Data = Buffer.from(jsonData, 'utf8').toString('base64');
+  const base64Origin = Buffer.from(appOrigin, 'utf8').toString('base64');
 
-  res.setHeader('Content-Type', 'text/html');
-  res.setHeader('X-Frame-Options', 'sameorigin');
+  // NOTE: The base64Data is essentially a raw UTF-8 byte stream. A string like
+  // 'ö' will be sent as 'w7Y=', which corresponds to the bytes C3 B6. If we do
+  // not UTF decode that on the receiving end, it'll turn out as 'Ã¶' instead
+  // of 'ö' when decoded by the UTF-unaware atob. We wrap the usage of
+  // Uint8Array and TextDecoder in a try/catch because, while their support is
+  // very widespread at the time of writing, they are not available on legacy
+  // IE browsers. The impact of not performing this step is not very grave; the
+  // only data that is not already 7-bit safe in the input data is generally in
+  // the profile info. So a name like 'Adelöw' may be displayed as 'AdelÃ¶w' on
+  // very old browsers.
+
+  // NOTE: We chose to do this seemingly roundabout encoding (instead of, for
+  // example, encodeURIComponent) because we prioritize safety over all else.
+  // If we used encodeURIComponent we would have to carefully use double quotes
+  // inside the script, because single quotes aren't encoded. Instead of
+  // risking inadvertently opening an attack surface when doing a seemingly
+  // innocuous change in the future, we choose an encoding that just cannot
+  // ever break the injected script.
 
   // TODO: Make target app origin configurable globally
   const script = `
-    (window.opener || window.parent).postMessage(JSON.parse(atob('${base64Data}')), '${appOrigin}')
-    window.close()
+    var json = atob('${base64Data}');
+    var origin = atob('${base64Origin}');
+    try {
+      var encoded = Uint8Array.from(Array.from(json, c => c.codePointAt(0)));
+      json = new TextDecoder().decode(encoded);
+    } catch (e) {
+    }
+    (window.opener || window.parent).postMessage(JSON.parse(json), origin);
+    window.close();
   `;
   const hash = crypto.createHash('sha256').update(script).digest('base64');
+
+  res.setHeader('Content-Type', 'text/html');
+  res.setHeader('X-Frame-Options', 'sameorigin');
```

Comment:
Setting this to `sameorigin` kinda defeats the purpose of being able to use this in an iframe if desired :p. Any particular reason you wanted to do it?

Reply:
I don't, this is just moved code from above (to keep all the `res.X` methods together)

Reply:
Shall I remove it as part of this PR? I assumed it was added for good reason

Reply:
Ah alright, nah keep it and we can see if we wanna change the behaviour in the future. I think the proper way to do it would be to explicitly allow the app origin but no other origins.